pkg/util/instancemetadata/prod.go

```diff
@@ -79,7 +80,53 @@ func (p *prod) populateTenantIDFromMSI(ctx context.Context) error {
 	return nil
 }
 
+// populateInstanceMetadataFromEnvironment attempts to populate instance metadata from environment, it does not throw errors.
+func (p *prod) populateInstanceMetadataFromEnvironment() bool {
+	environment := azure.PublicCloud
+	if value, found := os.LookupEnv("AZURE_ENVIRONMENT"); found {
+		var err error
+		environment, err = azure.EnvironmentFromName(value)
+		if err == nil {
+			p.environment = &environment
+		}
+	}
+
+	// Getenv returns empty string if not set which indicates no value for the struct
+	p.subscriptionID = os.Getenv("AZURE_SUBSCRIPTION_ID")
+	p.tenantID = os.Getenv("AZURE_TENANT_ID")
+	p.location = os.Getenv("LOCATION")
+	p.resourceGroup = os.Getenv("RESOURCEGROUP")
+
+	hostname, err := os.Hostname()
+	if err == nil {
+		p.hostname = hostname
+	}
```

Comment:
Will this not be a container hostname? OR this is what we want?

Reply:
Looking through usage it appears to be only logging and metrics.  We could pick from an env var as an option but not require it.  Fallback to this behavior.  I'll update.